src/hotspot/share/gc/g1/g1RemSet.cpp

```diff
@@ -555,24 +555,20 @@ class G1ScanHRForRegionClosure : public HeapRegionClosure {
       return (*card & ToScanMask) == 0;
     }
 
-    static bool is_word_aligned(const void* const addr) {
-      return ((uintptr_t)addr) % sizeof(Word) == 0;
-    }
-
     CardValue* find_first_dirty_card(CardValue* i_card) const {
-      while (!is_word_aligned(i_card)) {
+      while (!CardTable::is_word_aligned(i_card)) {
         if (is_card_dirty(i_card)) {
           return i_card;
         }
         i_card++;
       }
 
-      for (/* empty */; i_card < _end_card; i_card += sizeof(Word)) {
-        Word word_value = *reinterpret_cast<Word*>(i_card);
+      for (/* empty */; i_card < _end_card; i_card += sizeof(CardWord)) {
+        CardWord word_value = *reinterpret_cast<CardWord*>(i_card);
         bool has_dirty_cards_in_word = (~word_value & ExpandedToScanMask) != 0;
 
         if (has_dirty_cards_in_word) {
-          for (uint i = 0; i < sizeof(Word); ++i) {
+          for (uint i = 0; i < sizeof(CardWord); ++i) {
```

Comment:
This loop assumes _end_card is word aligned.  Should be asserted, or different loop bound.
Similarly in find_first_non_dirty_card.